### PR TITLE
Remove dead code (part 1)

### PR DESCRIPTION
**Pull request by mathemancer**
Related to #3946 

Removes dead code from the `mathesar` library.

**Technical details**

Main parts removed are:

- API tests which aren't relevant for the new architecture
- unused paths (in `urls.py`)
- Unused viewsets
- Unused serializers
- Unused API exception classes
- Dependencies of the above code (which aren't otherwise used)

This PR _does not_ remove:

- Any models, used or not
- Any code that supports models
- Any code from the `db` library.

## Checklist
 
- [X] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [X] My pull request targets the `develop` branch of the repository
- [X] My commit messages follow [best practices][best_practices].
- [X] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [X] I added or updated documentation (if applicable).
- [X] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by mathemancer** [comment]
@Anish9901 This is ready for review, but (again) don't merge until we cut the release branch.

File: mathesar/tests/data/data_types.csv
Comment (Anish9901) [inline mathesar/tests/data/data_types.csv:1]: I would like to have this file around in our codebase, I frequently use it to test import as well as test type inference manually. @mathemancer can you consider reverting this file in your subsequent PR?

**Review by Anish9901 — APPROVED** [review]
Rest in git history!

File: mathesar/tests/data/data_types.csv
Comment (mathemancer) [reply mathesar/tests/data/data_types.csv:1]: I see your point, but I'd like to get everything completely out first, then add things back in that we still want later. In that example, we'd want to have a separate directory where "human testing" data or files would reside.

I suggest making an issue for adding "human testing" data to the repo where we can discuss further.  (maybe we'd actually use the `mathesar-data-playground` repo or something)

**Pull request merged**